contracts/strategies/StrategyDAI3poolIron.sol

```diff
@@ -0,0 +1,562 @@
+        IERC20(cToken).safeTransfer(_strategy, IERC20(cToken).balanceOf(address(this)));
+        IERC20(y3crv).safeTransfer(_strategy, IERC20(y3crv).balanceOf(address(this)));
+        IERC20(_3crv).safeTransfer(_strategy, IERC20(_3crv).balanceOf(address(this)));
+        IERC20(want).safeTransfer(_strategy, IERC20(want).balanceOf(address(this)));
+    }
+
+    function forceD(uint256 _amount) external isAuthorized {
+        drip();
+        IERC20(want).safeApprove(_3pool, 0);
+        IERC20(want).safeApprove(_3pool, _amount);
+        uint256 v = _amount.mul(1e18).div(ICurveFi(_3pool).get_virtual_price());
+        ICurveFi(_3pool).add_liquidity([_amount, 0, 0], v.mul(DENOMINATOR.sub(slip)).div(DENOMINATOR));
+        if (_amount < tank) tank = tank.sub(_amount);
+        else tank = 0;
+
+        uint256 _bal = IERC20(_3crv).balanceOf(address(this));
+        IERC20(_3crv).safeApprove(y3crv, 0);
+        IERC20(_3crv).safeApprove(y3crv, _bal);
+        yvERC20(y3crv).deposit(_bal);
+
+        _bal = IERC20(y3crv).balanceOf(address(this));
+        IERC20(y3crv).safeApprove(cToken, 0);
+        IERC20(y3crv).safeApprove(cToken, _bal);
+        require(ICErc20(cToken).mint(_bal) == 0, "!forceD");
+        urn = ICErc20(cToken).balanceOfUnderlying(address(this));
+    }
+
+    function forceW(uint256 _amt) external isAuthorized {
+        drip();
+        require(ICErc20(cToken).redeemUnderlying(_amt) == 0, "!forceW");
+        urn = ICErc20(cToken).balanceOfUnderlying(address(this));
+        uint256 _before = IERC20(_3crv).balanceOf(address(this));
+        yvERC20(y3crv).withdraw(_amt);
+        uint256 _after = IERC20(_3crv).balanceOf(address(this));
+        _amt = _after.sub(_before);
+
+        IERC20(_3crv).safeApprove(_3pool, 0);
+        IERC20(_3crv).safeApprove(_3pool, _amt);
+        _before = IERC20(want).balanceOf(address(this));
+        ICurveFi(_3pool).remove_liquidity_one_coin(_amt, 0, _amt.mul(DENOMINATOR.sub(slip)).div(DENOMINATOR));
+        _after = IERC20(want).balanceOf(address(this));
+        tank = tank.add(_after.sub(_before));
+    }
+
+    function drip() public isAuthorized {
+        uint256 _p = yvERC20(y3crv).getPricePerFullShare();
+        _p = _p.mul(ICurveFi(_3pool).get_virtual_price()).div(1e18);
+        require(_p >= p, "backward");
+        uint256 _u = ICErc20(cToken).balanceOfUnderlying(address(this));
+        require(_u >= urn, "urn leak");
+        uint256 _r = ((_p.mul(_u)).sub(p.mul(urn))).div(1e18);
+        uint256 _s = _r.mul(strategistReward).div(DENOMINATOR);
+        uint256 _t = _r.mul(treasuryFee).div(DENOMINATOR);
+        uint256 _e = ICErc20(cToken).exchangeRateStored();
+        IERC20(cToken).safeTransfer(strategist, _s.mul(1e36).div(_p).div(_e));
+        IERC20(cToken).safeTransfer(Controller(controller).rewards(), _t.mul(1e36).div(_p).div(_e));
+        if (balanceOfy3CRV() > 0 && p > 0) {
+            _r = (_p.sub(p)).mul(balanceOfy3CRV()).div(1e18);
```

Comment:
minor gas saving by saving `balanceOfy3CRV()` in a var and use for operation after.